### PR TITLE
feat: 🎸 add SQFormCheckboxGroup and SQFormCheckboxGroupItem

### PR DESCRIPTION
**Pull request by lucas-homer**
SQFormCheckboxGroup and -Item comps enable an 'inclusion list' UX to
add/remove items from a larger list of options via checkbox, with an
optional 'select all' functionality built in.

[Loom demo](https://www.loom.com/share/021e1f7fed42443181b4285aa81cdf3f)

### CODE REVIEW

File: src/components/SQForm/SQFormCheckboxGroup.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          label="Select All"
+          isChecked={values.selectAll}
+          onChange={e => {
+            if (e.target.checked) {
+              setFieldValue(name, selectAllData);
+            } else {
+              setFieldValue(name, []);
+            }
+            setFieldValue('selectAll', !values.selectAll);
+          }}
```
Comment (20BBrown14) [inline src/components/SQForm/SQFormCheckboxGroup.js:38]: For readability sake could we pull this anon function out to a const?

File: src/components/SQForm/SQFormCheckboxGroup.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          {...selectAllProps}
+        />
+      </Grid>
+      <SQFieldArray name={name}>{children}</SQFieldArray>
+    </>
+  );
+}
+
+SQFormCheckboxGroup.propTypes = {
```
Comment (lucas-homer) [inline src/components/SQForm/SQFormCheckboxGroup.js:47]: forgot to add the other proptypes! standby for that...

File: src/components/SQForm/SQFormCheckboxGroup.js (hunk 1 above)
Comment (lucas-homer) [reply src/components/SQForm/SQFormCheckboxGroup.js:38]: sure! good idea.

Comment (lucas-homer) [reply src/components/SQForm/SQFormCheckboxGroup.js]: done

**Review by SeanGroff — APPROVED** [review]
This looks solid to me!
As we start using this more in the wild we may notice common patterns we can abstract into this component.
Since we don't know those yet, I think the current solution of going full-on inversion is solid

**Review by 20BBrown14 — APPROVED** [review]

**Pull request merged**